src/OpenTK/Input/GamePadConfigurationSource.cs

```diff
@@ -32,22 +32,23 @@ namespace OpenTK.Input
     struct GamePadConfigurationSource
     {
         int? map_button;
-        JoystickAxis? map_axis;
+        int? map_axis;
         JoystickHat? map_hat;
         HatPosition? map_hat_position;
 
-        public GamePadConfigurationSource(JoystickAxis axis)
+        public GamePadConfigurationSource(bool axis, int index)
```

Comment:
bool parameters are awful for readability. But this is mostly an internal class right? Would clients of OpenTK call this?
Maybe as a halfway point between this and factory methods make a small enum Source.Axis/Source.Button?